lib/pbench/agent/base.py

```diff
@@ -21,10 +21,14 @@ def __init__(self, context):
         self.config = PbenchAgentConfig(self.context.config)
         self.name = os.path.basename(sys.argv[0])
 
-        self.pbench_run = self.config.pbench_run
+        env_pbench_run = os.environ.get("pbench_run")
+        self.pbench_run = pathlib.Path(
+            self.config.pbench_run if env_pbench_run is None else env_pbench_run
+        )
+
         if not self.pbench_run.exists():
             click.secho(
-                f"[ERROR] the provided pbench run directory, {self.pbench_run}, does not exist"
+                f"[ERROR] pbench run directory does not exist, {self.pbench_run}"
             )
```

Comment:
Since you're not providing a "style", I suggest using [`echo()`](https://click.palletsprojects.com/en/8.0.x/api/#click.echo) instead of `secho()`.  Also, since this is intended to be an error message, I suggest sending it to `stderr` instead of the default `stdout` by providing the `err=true` parameter.)